source/plugins/ruby/in_kube_events.rb

```diff
@@ -141,6 +142,18 @@ def enumerate
         if (@eventsCount > 0)
           ApplicationInsightsUtility.sendMetricTelemetry("EventCount", @eventsCount, {})
         end
+
+        # Adding telemetry to send kubevents telemetry every 5 minutes
+        timeDifference = (DateTime.now.to_time.to_i - @@kubeEventsTelemetryTimeTracker).abs
+        timeDifferenceInMinutes = timeDifference / 60
+        if (timeDifferenceInMinutes >= 5)
+          telemetryFlush = true
+        end
+
+        if telemetryFlush
+          ApplicationInsightsUtility.sendCustomEvent("KubeEventsHeartBeatEvent", {})
```

Comment:
After emitting event, you would need to reset telemetryFlush = false, right? otherwise this will be emitted for every iteration.